### PR TITLE
[16.01] make dependencies browsable again

### PR DESCRIPTION
**Pull request by martenson**
it would be denied previously as the dependency folder is
outside of the repository (symlink protection)

this has milestone of 16.04 set so it will show up in the upcoming release notes

### CODE REVIEW

**Comment by martenson** [comment]
fixes #1834

**Comment by jmchilton** [comment]
Some tool shed tests are failing...

```
Error - <type 'exceptions.AttributeError'>: 'GalaxyWebTransaction' object has no attribute 'is_admin'
URL: http://localhost:8898/repository/get_file_contents?repository_id=adb5f5c93f827949&file_path=%2Fgalaxy%2Ftest%2Ftool_shed%2Ftmp%2Ftmpjyiaic%2Fdatabase%2Ffiles%2F000%2Frepo_1%2Ffiltering.xml
File '/galaxy/lib/galaxy/web/framework/middleware/error.py', line 151 in __call__
  app_iter = self.application(environ, sr_checker)
File '/galaxy/.venv/local/lib/python2.7/site-packages/paste/recursive.py', line 85 in __call__
  return self.application(environ, start_response)
File '/galaxy/.venv/local/lib/python2.7/site-packages/paste/httpexceptions.py', line 640 in __call__
  return self.application(environ, start_response)
File '/galaxy/lib/galaxy/web/framework/base.py', line 126 in __call__
  return self.handle_request( environ, start_response )
File '/galaxy/lib/galaxy/web/framework/base.py', line 184 in handle_request
  body = method( trans, **kwargs )
File '/galaxy/lib/galaxy/web/framework/decorators.py', line 48 in call_and_format
  return dumps( func( self, trans, *args, **kwargs ), **json_kwargs )
File '/galaxy/lib/galaxy/webapps/tool_shed/controllers/repository.py', line 1474 in get_file_contents
  is_admin = trans.is_admin
AttributeError: 'GalaxyWebTransaction' object has no attribute 'is_admin'
```

**Comment by martenson** [comment]
@jmchilton TS tests pass now

**Comment by nsoranzo** [comment]
`open_repository_files_folder()` function is also called once in the `open_folder()` method inside `lib/galaxy/webapps/tool_shed/controllers/repository.py`. I think something like:

```
        is_admin = trans.user_is_admin()
        return suc.open_repository_files_folder( trans.app, folder_path, repository_id )
```

should be used there.

File: lib/tool_shed/util/shed_util_common.py [hunk 1]
```diff
@@ -1150,9 +1149,18 @@ def open_repository_files_folder( app, folder_path, repository_id ):
     return folder_contents
 
 
+def is_path_browsable( app, path, repository_id, is_admin=False ):
```
Comment (nsoranzo) [inline lib/tool_shed/util/shed_util_common.py:1152]: Can you add a short function doc?

File: lib/tool_shed/util/shed_util_common.py [hunk 2]
```diff
@@ -1150,9 +1149,18 @@ def open_repository_files_folder( app, folder_path, repository_id ):
...
+    allowed = False
+    if is_admin and is_path_within_dependency_dir( app, path ):
+        allowed = True
+    if not allowed:
+        allowed = is_path_within_repo( app, path, repository_id)
+    return allowed
```
Comment (nsoranzo) [inline lib/tool_shed/util/shed_util_common.py]:
``` python
if is_admin and is_path_within_dependency_dir( app, path ):
    return True
return is_path_within_repo( app, path, repository_id)
```

**Comment by martenson** [comment]
@nsoranzo I incorporated your suggestions and fixed the bug you found. Thanks!

**Comment by nsoranzo** [comment]
:+1:

**Pull request merged**